### PR TITLE
IGNITE-18791 Add SSL Client authentication to REST

### PR DESCRIPTION
**Pull request by PakhomovAlexander**
https://issues.apache.org/jira/browse/IGNITE-18791

TCP Client and JDBC support this feature, now REST does it too.

### CODE REVIEW

File: modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java [hunk 1]
```diff
@@ -214,12 +217,18 @@ private int mapServerStartupException(ServerStartupException exception) {
     private Map<String, Object> properties(int port, int sslPort) {
         boolean dualProtocol = restConfiguration.dualProtocol().value();
         boolean sslEnabled = restConfiguration.ssl().enabled().value();
+        ClientAuth clientAuth = ClientAuth.valueOf(restConfiguration.ssl().clientAuth().value().toUpperCase());
```
[the remark this answers is not in the document]
Comment (PakhomovAlexander) [reply modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java:220]: Yes, there are safe. I've added the test for that case.

File: modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java [hunk 2]
```diff
@@ -228,11 +242,75 @@ private Map<String, Object> properties(int port, int sslPort) {
...
+            return result;
         } else {
             return Map.of("micronaut.server.port", port);
         }
     }
 
+    private static void validateKeyStorePath(String keyStorePath) {
+        if (keyStorePath.trim().isEmpty()) {
+            throw new IgniteException(
+                    Common.UNEXPECTED_ERR,
```
Comment (ptupitsyn) [inline modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java:279]: Here and below - I think we should use `ErrorGroups.Common.SSL_CONFIGURATION_ERR`.

File: modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java [hunk 3]
```diff
@@ -228,11 +242,75 @@ private Map<String, Object> properties(int port, int sslPort) {
...
+                    "Trust store path is not configured. Please check your rest.ssl.keyStore.path configuration."
+            );
+        }
+        if (!Files.exists(Path.of(keyStorePath))) {
```
Comment (ptupitsyn) [inline modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java:283]: Do we need those checks for existing file? Can Micronaut take care of this for us?

Comment (ptupitsyn) [inline modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java:283]: Here and below - please add a blank line between code blocks for readability (not sure why checkstyle does not catch this).

**Review by ptupitsyn — CHANGES_REQUESTED** [review]

File: modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java (hunk 3 above)
Comment (PakhomovAlexander) [reply modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java:283]: I've tried to leave this for micronaut but the error was weird and hard to understand.

Comment (ptupitsyn) [reply modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java:284]: Ok, got it.

**Review by ptupitsyn — APPROVED** [review]

**Pull request merged**